src/evaluators/BinaryExpression.js

```diff
@@ -74,7 +74,7 @@ export function getPureBinaryOperationResultType(
   } else if (op === ">>>" || op === "<<" || op === ">>" || op === "&" || op === "|" || op === "^" ||
              op === "**" || op === "%" || op === "/" || op === "*" || op === "-") {
     return reportErrorIfNotPure(IsToNumberPure, NumberValue);
-  } else if (op === "in") {
+  } else if (op === "in" || op === "instanceof") {
```

Comment:
It turns out that it is perfectly safe already. Far too safe, in fact. But making a better alternative to isSimple is best left to a later pull request.